app/controllers/users_controller.rb

```diff
@@ -0,0 +1,37 @@
+class UsersController < ApplicationController
+  def new
+    @user = User.new
+  end
+
+  def create
+    @user = User.new(user_params)
+    if @user.save
+      log_in @user
+      flash[:success] = "Welcome to Private Events!"
+      redirect_to @user
+    else
+      render 'new'
+      end
+  end
+
+  def show
+    @user = User.find(params[:id])
+    if logged_in?
+      @event = current_user.events.build
+      @feed_items = current_user.feed.paginate(page: params[:page])
+      @prev_events = []
+      @upcoming_events = []
+      @user.event_attendance.each do |x|
+        y = Event.find_by(id: x.attended_event)
+        @upcoming_events << y if y.date >= DateTime.now
+        @prev_events << y if y.date < DateTime.now
+      end
```

Comment:
1. It's the user's show page. So you should find the user from his id not only the `current_user`

2. In a few lines, 
```ruby
@prev_events = @user.events.prev_events
@upcoming_events = @user.events.upcoming_events
```